File: src/main/java/no/nav/aap/api/config/AbstractRestConfig.java
package no.nav.aap.api.config;

import java.net.URI;

import static no.nav.aap.api.util.URIUtil.uri;

public class AbstractRestConfig {

    private static final String IDPORTEN = "idporten";
    private final URI baseUri;
    private final String pingPath;
    private final boolean enabled;
    public static final String ISSUER = IDPORTEN;

    protected AbstractRestConfig(URI baseUri, String pingPath, boolean enabled) {
        this.baseUri = baseUri;
        this.pingPath = pingPath;
        this.enabled = enabled;
    }

    public URI pingEndpoint() {
        return uri(baseUri, pingPath);
    }

    public boolean isEnabled() {
        return enabled;
    }

    public URI getBaseUri() {
        return baseUri;
    }

    public String name() {
        return baseUri.getHost();
    }
}

File: src/main/kotlin/no/nav/aap/api/rest/AbstractRestConfig.kt
package no.nav.aap.api.rest

import no.nav.aap.api.util.URIUtil
import java.net.URI

 abstract class AbstractRestConfig(val baseUri: URI, protected val pingPath: String, val isEnabled: Boolean) {
    fun pingEndpoint() = URIUtil.uri(baseUri, pingPath)
    fun name()  = baseUri.host
     override fun toString() = "${javaClass.simpleName} [pingPath=$pingPath,enabled=$isEnabled,baseUri=$baseUri]"
 }